### PR TITLE
opensc: handle pcsc-lite conflict

### LINKED ISSUE #46109: opensc failed to build on 10.11.1
```bash
==> Installing opensc
==> Downloading https://downloads.sourceforge.net/project/opensc/OpenSC/opensc-0
==> Downloading from http://nchc.dl.sourceforge.net/project/opensc/OpenSC/opensc
######################################################################## 100.0%
==> Downloading https://github.com/carlhoerberg/OpenSC/commit/e5ae77cae32fdcc7a2
######################################################################## 100.0%
==> Patching
patching file configure.ac
==> ./bootstrap
==> ./configure --prefix=/usr/local/Cellar/opensc/0.15.0 --enable-sm --enable-op
Last 15 lines from /Users/agatti/Library/Logs/Homebrew/opensc/02.configure:
checking if more special flags are required for pthreads... -D_THREAD_SAFE
checking for PTHREAD_PRIO_INHERIT... yes
checking for inflate in -lz... yes
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
checking for readline in -lreadline... yes
checking readline/readline.h usability... yes
checking readline/readline.h presence... yes
checking for readline/readline.h... yes
checking for OPENSSL... yes
checking winscard.h usability... no
checking winscard.h presence... no
checking for winscard.h... no
configure: error: winscard.h is required for pcsc

READ THIS: https://git.io/brew-troubleshooting
```

Full logs here: https://gist.github.com/bb104c53655d12067b58

### PR DESCRIPTION
**Pull request by DomT4**
Will fix #46109 if it works, but since I have no readily-available system that has never been touched by a CLT package this needs double checking.

CC @agatti - If you wanted to pull this locally and test it, that'd be awesome.

### CODE REVIEW

**Comment by DomT4** [comment]
Apparently the test really doesn't like Yosemite.

**Comment by agatti** [comment]
Still same result on El Capitan:

```bash
agatti@tardigrade ~ $ brew install opensc
==> Installing dependencies for opensc: docbook, docbook-xsl
==> Installing opensc dependency: docbook
==> Downloading https://homebrew.bintray.com/bottles/docbook-5.0.el_capitan.bottle.3.tar.gz
######################################################################## 100.0%
==> Pouring docbook-5.0.el_capitan.bottle.3.tar.gz
==> xmlcatalog --noout --create /usr/local/etc/xml/catalog
==> xmlcatalog --noout --del file:///usr/local/Cellar/docbook/5.0/docbook/xml/4.2/catalog.xml /usr/local/etc/xml/catalog
==> xmlcatalog --noout --add nextCatalog  file:///usr/local/Cellar/docbook/5.0/docbook/xml/4.2/catalog.xml /usr/local/etc/xml/catalog
==> xmlcatalog --noout --del file:///usr/local/Cellar/docbook/5.0/docbook/xml/4.1.2/catalog.xml /usr/local/etc/xml/catalog
==> xmlcatalog --noout --add nextCatalog  file:///usr/local/Cellar/docbook/5.0/docbook/xml/4.1.2/catalog.xml /usr/local/etc/xml/catalog
==> xmlcatalog --noout --del file:///usr/local/Cellar/docbook/5.0/docbook/xml/4.3/catalog.xml /usr/local/etc/xml/catalog
==> xmlcatalog --noout --add nextCatalog  file:///usr/local/Cellar/docbook/5.0/docbook/xml/4.3/catalog.xml /usr/local/etc/xml/catalog
==> xmlcatalog --noout --del file:///usr/local/Cellar/docbook/5.0/docbook/xml/4.4/catalog.xml /usr/local/etc/xml/catalog
==> xmlcatalog --noout --add nextCatalog  file:///usr/local/Cellar/docbook/5.0/docbook/xml/4.4/catalog.xml /usr/local/etc/xml/catalog
==> xmlcatalog --noout --del file:///usr/local/Cellar/docbook/5.0/docbook/xml/4.5/catalog.xml /usr/local/etc/xml/catalog
==> xmlcatalog --noout --add nextCatalog  file:///usr/local/Cellar/docbook/5.0/docbook/xml/4.5/catalog.xml /usr/local/etc/xml/catalog
==> xmlcatalog --noout --del file:///usr/local/Cellar/docbook/5.0/docbook/xml/5.0/catalog.xml /usr/local/etc/xml/catalog
==> xmlcatalog --noout --add nextCatalog  file:///usr/local/Cellar/docbook/5.0/docbook/xml/5.0/catalog.xml /usr/local/etc/xml/catalog
==> Caveats
To use the DocBook package in your XML toolchain,
you need to add the following to your ~/.bashrc:

export XML_CATALOG_FILES="/usr/local/etc/xml/catalog"
==> Summary
🍺  /usr/local/Cellar/docbook/5.0: 181 files, 5.0M
==> Installing opensc dependency: docbook-xsl
==> Downloading https://homebrew.bintray.com/bottles/docbook-xsl-1.78.1_1.el_capitan.bottle.tar.gz
######################################################################## 100.0%
==> Pouring docbook-xsl-1.78.1_1.el_capitan.bottle.tar.gz
==> xmlcatalog --noout --del file:///usr/local/Cellar/docbook-xsl/1.78.1_1/docbook-xsl/catalog.xml /usr/local/etc/xml/catalog
==> xmlcatalog --noout --add nextCatalog  file:///usr/local/Cellar/docbook-xsl/1.78.1_1/docbook-xsl/catalog.xml /usr/local/etc/xml/catalog
==> xmlcatalog --noout --del file:///usr/local/Cellar/docbook-xsl/1.78.1_1/docbook-xsl-ns/catalog.xml /usr/local/etc/xml/catalog
==> xmlcatalog --noout --add nextCatalog  file:///usr/local/Cellar/docbook-xsl/1.78.1_1/docbook-xsl-ns/catalog.xml /usr/local/etc/xml/catalog
🍺  /usr/local/Cellar/docbook-xsl/1.78.1_1: 3698 files, 52M
==> Installing opensc
==> Downloading https://downloads.sourceforge.net/project/opensc/OpenSC/opensc-0.15.0/opensc-0.15.0.tar.gz
Already downloaded: /Library/Caches/Homebrew/opensc-0.15.0.tar.gz
==> Downloading https://github.com/carlhoerberg/OpenSC/commit/e5ae77cae32fdcc7a23d6bd0013c2fd115a43591.diff
Already downloaded: /Library/Caches/Homebrew/opensc--patch-18bd9b6220bfc03768c6a7f5324e7f3981eff0bc8b8f7eb0f5159508b43d6863.diff
==> Patching
patching file configure.ac
==> ./bootstrap
==> ./configure --prefix=/usr/local/Cellar/opensc/0.15.0 --enable-sm --enable-openssl --enable-pcsc --with-xsl-stylesheetsdir=/usr/local/opt/docbook-xsl/docbook-xsl
Last 15 lines from /Users/agatti/Library/Logs/Homebrew/opensc/02.configure:
checking if more special flags are required for pthreads... -D_THREAD_SAFE
checking for PTHREAD_PRIO_INHERIT... yes
checking for inflate in -lz... yes
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
checking for readline in -lreadline... yes
checking readline/readline.h usability... yes
checking readline/readline.h presence... yes
checking for readline/readline.h... yes
checking for OPENSSL... yes
checking winscard.h usability... no
checking winscard.h presence... no
checking for winscard.h... no
configure: error: winscard.h is required for pcsc

READ THIS: https://git.io/brew-troubleshooting

These open issues may also help:
opensc: fix no-CLT build https://github.com/Homebrew/homebrew/pull/46249
opensc failed to build on 10.11.1 https://github.com/Homebrew/homebrew/issues/46109
```

with logs at https://gist.github.com/fc88ba374359310caa94

Thank you for trying to fix this!

**Comment by agatti** [comment]
Manually launching configure with: ```PCSC_FLAGS=-I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/PCSC.framework/Headers ./configure --disable-dependency-tracking --prefix=/usr/local/ --enable-sm --enable-openssl --enable-pcsc``` did work, though:

```bash
checking winscard.h usability... yes
checking winscard.h presence... no
configure: WARNING: winscard.h: accepted by the compiler, rejected by the preprocessor!
configure: WARNING: winscard.h: proceeding with the compiler's result
checking for winscard.h... yes
```

Which is what you are doing in the formula, more or less.  Wonder why the environment variable doesn't get picked up when building...

**Comment by DomT4** [comment]
That's weird. What do the:

```
PC/SC default provider:  /System/Library/Frameworks/PCSC.framework/PCSC
PCSC_CFLAGS:             -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/PCSC.framework/Headers
```

Lines print for you if you do `make` after that successful `./configure`?

**Comment by agatti** [comment]
Hold on, it's picking up pcsc-lite's headers.  I'll remove the package and try again, sorry for that.  Incidentally, I think the patch at https://github.com/carlhoerberg/OpenSC/commit/e5ae77cae32fdcc7a23d6bd0013c2fd115a43591.diff might interfere with the PCSC_CFLAGS environment variable setting.

**Comment by DomT4** [comment]
Yeah, `PCSC_CFLAGS` is already set to that variable automatically inside upstream's `configure` script. My manually setting it was more or less curiosity to see if there were some issues with that logic that caused it to wander off in certain circumstances.

**Comment by agatti** [comment]
Building manually, _without_ applying the extra configure.ac patch, with:

```PCSC_CFLAGS=-I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/PCSC.framework/Headers ./configure --disable-dependency-tracking --prefix=/usr/local/Cellar/opensc/0.15.0 --enable-sm --enable-openssl --enable-pcsc```, yields:

```
PC/SC default provider:  /System/Library/Frameworks/PCSC.framework/PCSC
PCSC_CFLAGS:             -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/PCSC.framework/Headers
```

**Comment by DomT4** [comment]
This is a curious one. I wonder if somehow the upstream patch is forcing it to fallback on the incorrect CLT-required path or something. Does a vanilla `brew install opensc` still fail for you even with `pcsc-lite` installed?

**Comment by DomT4** [comment]
Sorry, even without `pcsc-lite`, is what I meant to say.

**Comment by agatti** [comment]
Believe it or not, without pcsc-lite and this patch it works.  Let me try again without the patch and I'll see if that was all a non-bug to begin with...

**Comment by DomT4** [comment]
Technically, still our bug even if it passes now for you locally because we should have done something about the `pcsc-lite` conflict, so :wink:

**Comment by agatti** [comment]
Yep, that was `pcsc-lite` interfering.  It built just fine with no patches: 

```
agatti@tardigrade ~/tmp/opensc-0.15.0 $ brew install opensc
==> Downloading https://downloads.sourceforge.net/project/opensc/OpenSC/opensc-0.15.0/opensc-0.15.0.tar.gz
Already downloaded: /Library/Caches/Homebrew/opensc-0.15.0.tar.gz
==> Downloading https://github.com/carlhoerberg/OpenSC/commit/e5ae77cae32fdcc7a23d6bd0013c2fd115a43591.diff
Already downloaded: /Library/Caches/Homebrew/opensc--patch-18bd9b6220bfc03768c6a7f5324e7f3981eff0bc8b8f7eb0f5159508b43d6863.diff
==> Patching
patching file configure.ac
==> ./bootstrap
==> ./configure --prefix=/usr/local/Cellar/opensc/0.15.0 --enable-sm --enable-openssl --enable-pcsc
==> make install
==> Caveats
Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
==> Summary
🍺  /usr/local/Cellar/opensc/0.15.0: 76 files, 6.9M, built in 55 seconds
```

**Comment by agatti** [comment]
Incidentally, if you need me to test other formulae in a CLT-less environment in the future, just ping me - I'll see what I can do for that :)

**Comment by DomT4** [comment]
Heh. Well, that's a relief of sorts; not having to spend too long poking around the build process is a nice upside for us both. I'll deal with the undeclared conflict here so others don't bump into it in future. Thanks for the assistance!
